rust/worker/src/execution/operators/knn.rs

```diff
@@ -26,6 +26,21 @@ impl PartialOrd for RecordDistance {
     }
 }
 
+/// The `KnnOperator` searches for the nearest neighbours of the specified embedding
+/// It can take different types of input and produce different types of output
+///
+/// # Parameters
+/// - `embedding`: The target embedding to search around
+/// - `fetch`: The number of records to fetch around the target
+///
+/// # Inputs
+/// In general, this operator takes inputs representing the domain of records to search within
```

Comment:
in general? what are the cases where it does not?

Reply:
I updated the comments to mention the two implementation of the `Operator<I, O>` trait for `KnnOperator`